playbooks/tasks/portal-renter-wait-until-idle.yml

```diff
@@ -0,0 +1,69 @@
+# Wait 5 minutes for any small uploads and downloads to finish. After 5 minutes,
+# it is likely that it is a large upload in which case it is safe to take the
+# server down as the large upload will continue on another server. Or it is a
+# large download which can also be resumed on another server.
+
+# Check if sia container is running.
+- name: Check sia container is running
+  community.docker.docker_container_info:
+    name: sia
+  register: sia_docker_container_result
+
+# Get the ip address for the sia container
+- name: Get sia service IP inside docker compose network
+  command: docker inspect -f "{% raw %}{{ range.NetworkSettings.Networks }}{{ .IPAddress }}{{ end }}{% endraw %}" sia
+  register: sia_ip_result
+
+- block:
+    - name: Set renter wait start time
+      set_fact:
+        renter_wait_start_time: "{{ (lookup('pipe', 'date +%s') | int) if renter_wait_start_time is not defined else renter_wait_start_time }}"
```

Comment:
IIRC `lookup('pipe')` runs on ansible control machine, not on host, times can be desynced.
I would use `ansible.builtin.command` and then `set_fact`.

Reply:
it doesn't matter if the time is not in sync with the machine, we need only to calculate delta since we want to make sure we wait up to 300s